src/examples/algorithms/Readme.md

```diff
@@ -0,0 +1,32 @@
+# C++ NuPIC HelloWorld example
+
+This file serves two purposes: 
+
+## Demo C++ aplication 
+
+Using only C++ `nupic.core` implementations, currently the chain uses 
+` SpatialPooler > TemporalPooler` 
+
+More classes should be added as they are ported (Encoders, Classifier, Anomaly, ...)
+
+## Profiling code 
+
+For ongoing optimization goals, we need a simple but complete code to run benchmarks and profile. 
+You can easily change the constants (`EPOCHS, DIM,...`) and try this code on your branch. 
+
+### Using `valgrind` profiler (for memory, #calls usage)
```

Comment:
Even better! When you do, please link it from the Developer Tips Wiki.

---------
Matt Taylor
OS Community Flag-Bearer
Numenta

On Mon, Mar 16, 2015 at 1:51 PM, breznak <notifications@github.com> wrote:

> In src/examples/algorithms/Readme.md
> <https://github.com/numenta/nupic.core/pull/388#discussion_r26509855>:
>
> > +
> > +This file serves two purposes:
> > +
> > +## Demo C++ aplication
> > +
> > +Using only C++ `nupic.core` implementations, currently the chain uses
> > +` SpatialPooler > TemporalPooler`
> > +
> > +More classes should be added as they are ported (Encoders, Classifier, Anomaly, ...)
> > +
> > +## Profiling code
> > +
> > +For ongoing optimization goals, we need a simple but complete code to run benchmarks and profile.
> > +You can easily change the constants (`EPOCHS, DIM,...`) and try this code on your branch.
> > +
> > +### Using `valgrind` profiler (for memory, #calls usage)
>
> I plan to start a Profiling/Optimizations wiki page soon...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numenta/nupic.core/pull/388/files#r26509855>.
>